scc/light_client/light_client.go

```diff
@@ -64,3 +67,45 @@ func (c *LightClient) Close() {
 func (c *LightClient) Sync() (idx.Block, error) {
 	return c.state.sync(c.provider)
 }
+
+// getAccountProof retrieves and verifies the proof for the given address.
+// Returns an error if the light client has not been synced or if the proof cannot be obtained.
+func (c *LightClient) getAccountProof(address common.Address) (carmen.WitnessProof, error) {
+	if !c.state.hasSynced {
+		return nil, fmt.Errorf("light client has not yet synced")
+	}
+	proof, err := c.provider.getAccountProof(address, c.state.headNumber)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get account proof: %w", err)
+	}
+	if proof == nil {
+		return nil, fmt.Errorf("nil account proof for address %v", address)
+	}
+	return proof, nil
+}
+
+// GetBalance returns the balance of the given address.
+// It returns an error if the light client has not been synced,
+// the balance could not be proven or there was any error
+// in getting or verifying the proof.
+func (c *LightClient) GetBalance(address common.Address) (*uint256.Int, error) {
+	if !c.state.hasSynced {
+		return nil, fmt.Errorf("light client has not yet synced")
+	}
```

Comment:
nit: redundant, since it is also checked inside the `getAccountProof` call following right after.

Reply:
https://github.com/0xsoniclabs/sonic/pull/145/commits/7d809671c18e57960b435da8d81a12baf2d0737c